### ISSUE TITLE
no-boolean-literal-compare should only work with bool type

### ISSUE THREAD

**Issue by nikitadol** [dart-code-checker/dart-code-metrics#266]
* **Dart code metrics version:** 3.0.0
* **Dart sdk version:** 2.12.2 (stable) (Wed Mar 17 10:30:20 2021 +0100) on "macos_x64"

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 20
    lines-of-executable-code: 50
    number-of-parameters: 5
    maximum-nesting-level: 5
  metrics-exclude:
    - 'test/**'
  rules:
    - avoid-returning-widgets
    - newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    - no-equal-then-else
    - prefer-conditional-expressions
    - prefer-trailing-comma

```

</details>

**What did you do? Please include the source code example causing the issue.**
If no parameter is specified, then the `push` method returns `dynamic`

```dart
    final res = await Navigator.of(context).push(...);
    
    if (res == true) {
```

**What did you expect to happen?**
This does not trigger the linter rule

**What actually happened?**
Linter shows warning
![Screenshot 2021-04-14 at 10 02 32](https://user-images.githubusercontent.com/16593680/114668891-ee1bd500-9d09-11eb-91cd-68a4a95bdb04.png)

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by dkrutskikh** [comment]
@nikitadol thank you for feedback. I will try to fix this issue in upcoming release
